### PR TITLE
feat: [IDP]: Added usage`formData` to pre-fill workflows

### PR DESCRIPTION
**Pull request by Debanitrkl**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/internal-developer-portal/flows/service-onboarding-pipelines.md [hunk 1]
```diff
@@ -1012,6 +1012,58 @@ The `projectSlug` filter generates a project slug from a repository URL
...
 
+## Pre-fill workflows with URL Params
+
+We can now automatically load IDP Workflow forms pre-filled using the `formData` URL query parameter. eg: `https://app.harness.io/ng/account/account_id/module/idp/create/templates/default/a-python-lambda?formData=%7B%22project_name%22%3A%22auto%20filled%22%7D`
+
+The query parameters `?formData=%7B%22project_name%22%3A%22auto%20filled%22%7` in the end of the URL allow you to automatically fill in values of the form. Please see the below table for explanation of individual tokens in the query param.
+
+| Item                | Example Value                           | Explanation                                                                                      |
+|---------------------|-----------------------------------------|--------------------------------------------------------------------------------------------------|
+| `formData`          | `formData`                              | Key of the query param.`formData` object is used to fill out scaffolder template forms.           |
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/flows/service-onboarding-pipelines.md:1023]:
```suggestion
| `formData`          | `formData`                              | Key of the query param.`formData` object is used to fill out IDP Workflow forms.           |
```

File: docs/internal-developer-portal/flows/service-onboarding-pipelines.md [hunk 2]
```diff
@@ -1012,6 +1012,58 @@ The `projectSlug` filter generates a project slug from a repository URL
...
+| `{"key"%3A"value"}` | `{"title"%3A"Title from query params"}` | Value of the query param. A JSON object with invalid URL characters encoded.`:` encodes to `%3A` |
+
+### Additional information
+
+Using automatically filled out values is handy when wanting to direct users to use scaffolder templates with known good values. This also allows automation to be constructed around the scaffolder, where the automation can provide fully constructed scaffolder URLs to the user. You can also prevent user from modifying the form values inserted from query params by making the form fields `readonly`. See below example of a minimal form which would be filled using query params defined in the above explanation.
+
+```YAML
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/flows/service-onboarding-pipelines.md:1030]: We should start making these YAML blocks click to expand kind of thing using <details> and <summary>. The page is getting extremely long and full of code blocks.

**Review by OrkoHunter — APPROVED** [review]

File: docs/internal-developer-portal/flows/service-onboarding-pipelines.md [hunk 3]
```diff
@@ -1012,6 +1012,58 @@ The `projectSlug` filter generates a project slug from a repository URL
...
+|---------------------|-----------------------------------------|--------------------------------------------------------------------------------------------------|
+| `formData`          | `formData`                              | Key of the query param.`formData` object is used to fill out scaffolder template forms.           |
+| `{"key"%3A"value"}` | `{"title"%3A"Title from query params"}` | Value of the query param. A JSON object with invalid URL characters encoded.`:` encodes to `%3A` |
+
+### Additional information
+
+Using automatically filled out values is handy when wanting to direct users to use scaffolder templates with known good values. This also allows automation to be constructed around the scaffolder, where the automation can provide fully constructed scaffolder URLs to the user. You can also prevent user from modifying the form values inserted from query params by making the form fields `readonly`. See below example of a minimal form which would be filled using query params defined in the above explanation.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/flows/service-onboarding-pipelines.md:1028]:
```suggestion
Using automatically filled out values is handy when wanting to direct users to use IDP Workflows with known good values. This also allows automation to be constructed around the Workflows, where the automation can provide fully constructed IDP URLs to the user. You can also prevent user from modifying the form values inserted from query params by making the form fields `readonly`. See below example of a minimal form which would be filled using query params defined in the above explanation.
```

Comment (OrkoHunter) [reply docs/internal-developer-portal/flows/service-onboarding-pipelines.md:1028]: Let's be consistent. Use more of
* IDP
* Workflows
* Forms
* Pipelines
* Step

Less of
* Scaffolder
* template
* action

**Review by sandeepa-kv-jois — DISMISSED** [review]

**Pull request merged**